contents/tutorials/monitor-llama-index-with-langfuse.md

```diff
@@ -0,0 +1,197 @@
+from llama_index.core import Settings
+
+# Add these lines of code just below the above existing imports
+from llama_index.core import Settings
+from llama_index.core.callbacks import CallbackManager
+from langfuse.llama_index import LlamaIndexCallbackHandler
+
+langfuse_callback_handler = LlamaIndexCallbackHandler()
+Settings.callback_manager = CallbackManager([langfuse_callback_handler])
+
+# Rest of your existing code...
+```
+
+## 3. Build RAG on the hedgehog pdf
+
+First download the [hedgehog care guide pdf](https://www.pro-igel.de/downloads/merkblaetter_engl/wildtier_engl.pdf) by running the following command in your project directory:
+
+```bash
+!wget 'https://www.pro-igel.de/downloads/merkblaetter_engl/wildtier_engl.pdf' -O './hedgehog.pdf'
+```
+
+Next, we load the pdf using the LlamaIndex [`SimpleDirectoryReader`](https://docs.llamaindex.ai/en/stable/module_guides/loading/simpledirectoryreader/), create vector embeddings using [`VectorStoreIndex`](https://docs.llamaindex.ai/en/stable/module_guides/indexing/vector_store_index/), and then convert it into a [queryable engine](https://docs.llamaindex.ai/en/stable/module_guides/deploying/query_engine/) that we can retrieve information from.
+
+```python
+# Add this to the bottom of app.py
+from llama_index.core import SimpleDirectoryReader
+
+hedgehog_docs = SimpleDirectoryReader(
+    input_files=["./hedgehog.pdf"]
+).load_data()
+
+from llama_index.core import VectorStoreIndex
+
+hedgehog_index = VectorStoreIndex.from_documents(hedgehog_docs)
+hedgehog_query_engine = hedgehog_index.as_query_engine(similarity_top_k=5)
+```
+
+Finally, we can add code to query the engine and print the answer:
+
+```python
+# Add this to the bottom of app.py
+response = hedgehog_query_engine.query("What is the best way to care for a hedgehog?")
+print(response)
+```
+
+Test the app by running `python app.py`. You should hopefully see a response like the one below 🎉: 
+
+```
+The best way to care for a hedgehog is to provide it with a suitable 
+habitat that includes nesting possibilities and a food supply located closely 
+together within a small area. This can be achieved by creating a garden 
+with a variety of natural structures such as hedges, bushes, groundcover 
+plants, piles of leaves, and undergrowth.
+```
+
+All steps of the LLM chain are now tracked in Langfuse and you can view them in your [Langfuse dashboard](https://cloud.langfuse.com/).
+
+<ProductScreenshot
+    imageLight = "https://res.cloudinary.com/dmukukwp6/image/upload/Screenshot_2024_12_06_at_1_29_21_PM_80ce94fb98.png" 
+    imageDark = " https://res.cloudinary.com/dmukukwp6/image/upload/Screenshot_2024_12_06_at_1_30_43_PM_fc294d1faf.png"
+    classes="rounded"
+    alt="Langfuse trace"
+/>
+
+## 4. Integrate Langfuse with PostHog
+
+Next we connect Langfuse to PostHog so that you can combine your LLM trace data with your PostHog analytics. This enables to answer product questions such as:
```

Comment:
```suggestion
Next we connect Langfuse to PostHog so that you can combine your LLM trace data with your PostHog analytics. This enables to answers to product questions such as:
```